compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -66,9 +66,44 @@ object Objects:
 
   // ----------------------------- abstract domain -----------------------------
 
+  /** Syntax for the data structure abstraction used in abstract domain:
+   *
+   * ve ::= ObjectRef(class)
+   *      | OfClass(class, vs[outer], ctor, args, env)
+   *      | OfArray(object[owner], regions)
+   *      | Fun(..., env)
+   *      | Cold                                                         // abstract values in domain
+   * vs ::= Set(ve)                                                      // set of abstract values
+   *
+   * refMap = ( ObjectRef | OfClass ) -> ( valsMap, varsMap, outersMap ) // refMap stores field informations of an object or instance
+   * valsMap = valsym -> vs                                              // maps immutable fields to their values
+   * varsMap = valsym -> addr                                            // each mutable field has an abstract address
+   * outersMap = class -> vs                                             // maps outer objects to their values
+   *
+   * arrayMap = OfArray -> addr                                          // an array has one address that stores the join value of every element
+   *
+   * heap = addr -> vs                                                   // only this map's entries can be modified
```

Comment:
The comment is not easy to understand. Do we mean that the heap should be immutable?
```suggestion
   * heap = addr -> vs                                                   // the map is immutable (see the optimization in implementation)
```